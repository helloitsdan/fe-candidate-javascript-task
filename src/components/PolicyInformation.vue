<template>
  <div class="columns">
    <div class="column is-one-third">
      <h4>Started on</h4>
      {{ start_date | moment("dddd, MMMM Do YYYY") }}
    </div>
    <div class="column is-one-third">
      <h4>Ends on</h4>
      {{ end_date | moment("dddd, MMMM Do YYYY") }}
    </div>
    <div class="column is-one-third">
      <h4>Status</h4>
      {{ status }}
    </div>
  </div>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  name: "PolicyInformation",
  props: {
    uuid: String,
    status: String,
    start_date: String,
    end_date: String
  }
});
</script>
