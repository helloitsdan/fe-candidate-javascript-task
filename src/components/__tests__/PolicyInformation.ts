import { shallowMount } from "@vue/test-utils";

import PolicyInformation from "../PolicyInformation.vue";

describe("PolicyInformation", () => {
  const fakePolicy = {
    uuid: "1234",
    status: "ACTIVE",
    start_date: "2019-07-13T02:57:55.235Z",
    end_date: "2022-01-10T12:32:11.195Z"
  };

  it("renders", () => {
    const wrapper = shallowMount(PolicyInformation, {
      propsData: fakePolicy
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
