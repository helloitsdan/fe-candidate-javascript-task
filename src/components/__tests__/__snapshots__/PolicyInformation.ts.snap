// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyInformation renders 1`] = `
<div
  class="columns"
>
  <div
    class="column is-one-third"
  >
    <h4>
      Started on
    </h4>
    
    Saturday, July 13th 2019
  
  </div>
   
  <div
    class="column is-one-third"
  >
    <h4>
      Ends on
    </h4>
    
    Monday, January 10th 2022
  
  </div>
   
  <div
    class="column is-one-third"
  >
    <h4>
      Status
    </h4>
    
    ACTIVE
  
  </div>
</div>
`;
